src/anyvar/storage/abc.py

```diff
@@ -0,0 +1,132 @@
+"""Provide PostgreSQL-based storage implementation."""
+
+import enum
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+
+from ga4gh.vrs import models as vrs_models
+
+
+class StoredObjectType(enum.StrEnum):
+    """Supported VRS object types for AnyVar storage."""
+
+    ALLELE = "Allele"
+    LOCATION = "Location"
+    COPY_NUMBER_COUNT = "CopyNumberCount"
+    COPY_NUMBER_CHANGE = "CopyNumberChange"
+    SEQUENCE_LOCATION = "SequenceLocation"
+    SEQUENCE_REFERENCE = "SequenceReference"
+
+
+class VariationMappingType(enum.StrEnum):
+    """Supported mapping types between VRS Variations."""
+
+    LIFTOVER = "liftover"
+    TRANSCRIPTION = "transcription"
+    TRANSLATION = "translation"
+
+
+class _Storage(ABC):
+    """Abstract base class for interacting with storage backends."""
+
+    @abstractmethod
+    def setup(self) -> None:
+        """Set up the storage backend."""
+
+    @abstractmethod
+    def close(self) -> None:
+        """Close the storage backend."""
+
+    @abstractmethod
+    def wait_for_writes(self) -> None:
+        """Wait for all background writes to complete.
+        NOTE: This is a no-op for synchronous storage backends.
+        """
+
+    @abstractmethod
+    def wipe_db(self) -> None:
+        """Wipe all data from the storage backend."""
+
+    @abstractmethod
+    def add_objects(self, objects: Iterable[vrs_models.VrsType]) -> None:
+        """Add multiple VRS objects to storage."""
+
+    @abstractmethod
+    def get_objects(
+        self, object_type: StoredObjectType, object_ids: Iterable[str]
+    ) -> Iterable[vrs_models.VrsType]:
+        """Retrieve multiple VRS objects from storage by their IDs."""
+
+    @abstractmethod
+    def get_all_object_ids(self) -> Iterable[str]:
+        """Retrieve all object IDs from storage."""
+
+    @abstractmethod
+    def get_object_count(self, object_type: StoredObjectType) -> int:
+        """Get count of objects of a specific type in storage."""
```

Comment:
For some reason I thought we'd agreed to remove this? Was that just wishful thinking?

Reply:
remove get_object_count